packages/opentelemetry-basic-tracer/src/BasicTracer.ts

```diff
@@ -0,0 +1,156 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as types from '@opentelemetry/types';
+import { BaseScopeManager } from '@opentelemetry/scope-base';
+import {
+  ALWAYS_SAMPLER,
+  NoopLogger,
+  NoopSpan,
+  randomSpanId,
+  randomTraceId,
+} from '@opentelemetry/core';
+import { BasicTracerConfig } from '../src/types';
+
+/**
+ * This class represents a basic tracer.
+ */
+export class BasicTracer implements types.Tracer {
+  static INVALID_ID = '0';
+
+  private defaultAttributes: types.Attributes;
+  private logger: types.Logger;
+  private sampler: types.Sampler;
+  private scopeManager?: BaseScopeManager;
+
+  // TODO: consume invalid span context from `SpanContext.INVALID`
+  static defaultSpan = new NoopSpan({
+    traceId: BasicTracer.INVALID_ID,
+    // TODO: consume invalid `spanId` from `Span.INVALID`
+    spanId: '0',
+    traceOptions: types.TraceOptions.UNSAMPLED,
+  });
+
+  /**
+   * Constructs a new Tracer instance.
+   */
+  constructor(config: BasicTracerConfig = {}) {
+    this.sampler = config.sampler || ALWAYS_SAMPLER;
+    this.scopeManager = config.scopeManager;
+    this.defaultAttributes = config.defaultAttributes || {};
+    this.logger = config.logger || new NoopLogger();
+  }
+
+  /**
+   * Starts a new Span or returns the default NoopSpan based on the sampling
+   * decision.
+   */
+  startSpan(name: string, options: types.SpanOptions = {}): types.Span {
+    let parentSpanContext: types.SpanContext | undefined;
+
+    // parent is a SpanContext
+    if (
+      options.parent &&
+      typeof (options.parent as types.SpanContext).traceId
```

Comment:
How is this converted to JavaScript? It doesn't look like this check is JS compatible. Same for other occurrences of `as`.

Reply:
`as` is just used to force the type of a variable, in this case it would be transpiled to `if (options.parent && options.parent.traceId) {}`

Reply:
Can we remove the `typeof` then?

Reply:
could someone show me an example of how would it work with TS?
I get a Type error without `typeof`, `instanceof` also doesn't work.

Reply:
What error do you get if you remove the `typeof`?

Reply:
Okay, I see now. The `typeof` can be removed, the `as`  not.